### PR TITLE
Feat: Simplex

### PR DESCRIPTION
**Pull request by Gamboster**

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 1]
```diff
@@ -946,6 +946,36 @@ export class ExpressApp {
       });
     });
 
+    router.post('/service/simplex/quote', (req, res) => {
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/expressapp.ts]: add `/v1/` to urls.

File: packages/bitcore-wallet-service/src/config.ts [hunk 2]
```diff
@@ -73,6 +73,18 @@ module.exports = {
   maintenanceOpts: {
     maintenanceMode: false,
   },
+  simplex: {
+    sandbox: {
+      apiKey: 'simplex_sandbox_api_key_here',
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/config.ts]: maybe better to keep this commented out. And is they are undef, make the code to ignore the related requests.

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 3]
```diff
@@ -4508,6 +4510,121 @@ export class WalletService {
 
     this.storage.removeTxConfirmationSub(this.copayerId, opts.txid, cb);
   }
+
+  simplexGetQuote(req): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (!req.body.env || (req.body.env != 'sandbox' && req.body.env != 'production')) return reject(new Error('Simplex\'s request wrong environment'));
+
+      const API = config.simplex[req.body.env].api;
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts]: handle the case config.simplex is not defined.

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 4]
```diff
@@ -4508,6 +4510,121 @@ export class WalletService {
...
+      const API_KEY = config.simplex[req.body.env].apiKey;
+      const ip = req.headers['x-real-ip'] || req.connection.remoteAddress || req.ip;
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts]: checking the code of `express-rate-limit.js` it seems `const ip = req.ip` is enough

File: packages/bitcore-wallet-service/src/lib/expressapp.ts (hunk 1 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/expressapp.ts]: Done

File: packages/bitcore-wallet-service/src/config.ts (hunk 2 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/config.ts]: Done

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 3 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: Done

**Comment by Gamboster** [comment]
> please add tests.

Tests in WIP

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 4 above)
Comment (Gamboster) [reply packages/bitcore-wallet-service/src/lib/server.ts]: I was reading that there are different ways in which the ip can come. I did a utils function that covers all cases

**Pull request merged**